lib/testing/assert.fnl

```diff
@@ -18,6 +18,27 @@
                  (+ matches 1)
                  matches))) (.. message " instead got " (hs.inspect actual))))
 
+(fn all? [pred iter]
+  (var result true)
+  ;; Properly iterate over the `pairs` or iterator
+  (each [k v (iter)]
+    (when (not (pred k v))
+      (set result false)))
```

Comment:
This is the 'slow way' since we don't exit early.

I could do `(lua "return false")`, but I am not sure if we'd like to, since it's not as lispy

Reply:
Would a while loop work well for this?

Reply:
It could be done, but `while` loops don't take iterators, so I would have to somehow break it into a list.

Reply:
https://fennel-lang.org/reference#accumulate-iterator-accumulation
Just remembered the accumulate family of macros accepts an `&until` clause. Maybe that would work?